symfit/core/minimizers.py

```diff
@@ -361,18 +376,43 @@ def scipy_constraints(self, constraints):
         """
         Returns all constraints in a scipy compatible format.
 
+        :param constraints: List of either MinimizeModel instances (this is what
+            is provided by :mod:`~symfit.core.fit.Fit`),
+            :mod:`~symfit.core.fit.Constraints`, or :mod:`~sympy.Rel`.
         :return: dict of scipy compatible statements.
         """
         cons = []
         types = {  # scipy only distinguishes two types of constraint.
             sympy.Eq: 'eq', sympy.Ge: 'ineq',
         }
 
-        for minimize_constraint in constraints:
-            constraint_type = minimize_constraint.model.constraint_type
+        for constraint in constraints:
+            if isinstance(constraint, MinimizeModel):
```

Comment:
I'd argue that if you want to call a Minimizer directly, you should do all the packaging and wrapping of input yourself. Especially since we disrecommend doing that in the first place. We can still consider providing a helper function that does it for you (which Fit also uses), but that would be pure benevolence on our part.

Reply:
So that would mean that we just assume a `MinimizeModel` instance here for constraints, as `Fit` provides, and take it from there?

I do tend to agree, but I also do think that the code example in #156 should work, it just looks so natural, don't you think? But even if we keep that one, there's one of the conditions here we could skip. I also definitely agree that people should be encouraged as much as possible to just use Fit.

Reply:
The more I think about it, the stronger I am of the opinion that if people choose not to use `Fit`, they should wrap their own things. That way 1) our code stays cleaner, since we *know* what kind of input we get, and 2) people still have access to the powertools. 
The way you propose here provides no additional *functionality* over people calling Fit, or wrapping their constraints in a sane way.

Reply:
But of course, we could provide some convenience functions to help them